src/asciidoctor-extensions-api.js

```diff
@@ -87,12 +95,51 @@ Extensions.unregisterAll = function () {
   this.$unregister_all();
 };
 
+/**
+ * Unregister the specified statically-registered extension groups.
+ *
+ * NOTE Opal cannot delete an entry from a Hash that is indexed by symbol, so
+ * we have to resort to using low-level operations in this method.
+ *
+ * @memberof Extensions
+ */
+Extensions.unregister = function () {
+  var names = Array.isArray(arguments[0]) ? arguments[0] : arguments;
+  var groups = this.groups;
+  var groupNameIdx = {};
+  for (var i = 0, groupSymbolNames = groups.$$keys, l1 = groupSymbolNames.length; i < l1; i++) {
```

Comment:
Does it really make a difference to cache the length of the array ?
In theory it should be faster but it introduces a new variable and in my opinion the code is a bit less readable.

I did search on the Internet but nothing conclusive... Also this array will most likely contains less than one hundred elements, so I think we are fine :rocket:

Reply:
You're correct the performance difference isn't measurable. This template is just hardwired in my head.